code/datums/station_traits/neutral_traits.dm

```diff
@@ -75,3 +75,77 @@
 /datum/station_trait/announcement_medbot/New()
 	. = ..()
 	SSstation.announcer = /datum/centcom_announcer/medbot
+
+/datum/station_trait/protagonist
+	name = "Announcement \"System\""
+	trait_type = STATION_TRAIT_NEUTRAL
+	weight = 0
+	show_in_report = TRUE
+	trait_flags = STATION_TRAIT_ABSTRACT
+	report_message = "This station has received an esteemed guest! They will most likely be a high value target for any Syndicate invasions. Be sure to keep them safe!"
+	blacklist = list()
+	///What role to give to the picked player
+	var/datum/antagonist/role_to_give
+	//The mind of the player that was picked
+	var/datum/mind/picked_mind
+	//The antag datum instance for the protagonist we are creating
+	var/datum/antagonist/antag_datum_instance
+
+
+/datum/station_trait/protagonist/New()
+	. = ..()
+	RegisterSignal(SSdcs, COMSIG_GLOB_PRE_GAMEMODE_SETUP, .proc/on_gamemode_setup)
+	RegisterSignal(SSdcs, COMSIG_GLOB_POST_GAMEMODE_SETUP, .proc/after_gamemode_setup)
+	antag_datum_instance = new role_to_give() //Create this early for things such as family name to be setup
+
+/// Checks if candidates are connected and if they are banned or don't want to be the antagonist.
+/datum/station_trait/protagonist/proc/trim_candidates(list/candidates)
+	for(var/mob/dead/new_player/candidate_player in candidates)
```

Comment:
```suggestion
	for(var/mob/dead/new_player/candidate_player as anything in candidates)
```
assuming that list/candidates is a list with only new players in it, adding "as anything" makes byond not try and filter the list for only candidate_player's type